pkg/kubelet/kubelet_pods_test.go

```diff
@@ -3781,6 +3781,101 @@ func Test_generateAPIPodStatus(t *testing.T) {
 	}
 }
 
+func Test_generateAPIPodStatusForInPlaceVPAEnabled(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScaling, true)()
+	testContainerName := "ctr0"
+	testContainerID := kubecontainer.ContainerID{Type: "test", ID: testContainerName}
+
+	CPU1AndMem1G := v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}
```

Comment:
I see so many  v1.ResourceList and clone replications. I would suggest you to create a inner function method to accept cpu, memory, ephemeral storage and return a resource list to make code cleaner and readable